internal/pkg/vm/actor/builtin/power/power_test.go

```diff
@@ -0,0 +1,190 @@
+package power_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	th "github.com/filecoin-project/go-filecoin/internal/pkg/testhelpers"
+	tf "github.com/filecoin-project/go-filecoin/internal/pkg/testhelpers/testflags"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/types"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/abi"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/actor"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/actor/builtin"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/actor/builtin/miner"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/actor/builtin/power"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/address"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/state"
+)
+
+func TestPowerCreateStorageMiner(t *testing.T) {
+	tf.UnitTest(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	st, vms := th.RequireCreateStorages(ctx, t)
+
+	pid := th.RequireRandomPeerID(t)
+	pdata := actor.MustConvertParams(address.TestAddress, address.TestAddress, pid, types.OneKiBSectorSize)
+	msg := types.NewUnsignedMessage(address.TestAddress, address.PowerAddress, 0, types.NewAttoFILFromFIL(100), power.CreateStorageMiner, pdata)
+	result, err := th.ApplyTestMessage(st, vms, msg, types.NewBlockHeight(0))
```

Comment:
argh wish I knew about this before :( 
I'll see what I can do.

Reply:
One thing to keep in mind is that the `ApplyTestMessage` pattern should probably be used once in at least every actor's test because it helps root out issues that the fake vmcontext tests can't.  

For example I caught errors from not setting the address to actor mapping in the genesis init function.  Without `ApplyTestMessage` I wouldn't have caught these until trying to use power actor messages in an integration test.

Reply:
That's definitely true in this case. Eventually I'd like to require FakeVMContext (or similar) unit test coverage plus at least happy path integration testing on all actor functionality.